src/main/java/reposense/model/RepoLocation.java

```diff
@@ -51,31 +59,97 @@ public String getOrganization() {
         return organization;
     }
 
+    public Optional<String> getParsedBranch() {
+        return parsedBranch;
+    }
+
     /**
-     * Verifies {@code location} can be presented as a {@code URL} or {@code Path}.
-     * @throws InvalidLocationException if otherwise.
+     * Given a String representing a repo location, returns
```

Comment:
I don't think the JavaDoc highlights what the method does. It just informs what it gets and what it returns - not sure if that implies what it does.

Reply:
I've used the verb `Parses` instead of `Given` and added a bit of information about the repo location. Do you think it is suitable now?